interfaces/api/v1/blueprint/continuous_planning_routes.py

```
@@ -108,19 +114,58 @@ async def generate_macro_plan(
-        return result
+        service.initialize_macro_plan_task(novel_id)
+
+        async def _generate_task():
+            try:
+                result = await service.generate_macro_plan(
+                    novel_id=novel_id,
+                    target_chapters=request.target_chapters,
+                    structure_preference=request.structure.dict()
+                )
+                service.store_macro_plan_result(novel_id, result)
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)